sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -762,15 +679,7 @@ class TungstenAggregationIterator(
   /**
    * Start processing input rows.
    */
-  testFallbackStartsAt match {
-    case None =>
-      processInputs()
-    case Some(fallbackStartsAt) =>
-      // This is the testing path. processInputsWithControlledFallback is same as processInputs
-      // except that it switches to sort-based aggregation after `fallbackStartsAt` input rows
-      // have been processed.
-      processInputsWithControlledFallback(fallbackStartsAt)
-  }
+  processInputs(testFallbackStartsAt.getOrElse(Int.MaxValue))
```

Comment:
I guess it isn't a huge deal to have a really high limit here, as opposed to no limit, because the cost of spilling after processing a couple of billion rows will be fairly minimal; either we would have spilled earlier or we're experiencing a huge reduction factor, so the cost of spilling should be minimal.